viewing_party/party.py

```diff
@@ -1,23 +1,168 @@
+    total = 0
+
+    #make sure the dictionary isn't empty
+    if user_data["watched"]:
+
+        #iterate over the watched list and add averages to total
+        for i in range(len(user_data["watched"])):
+            total += user_data["watched"][i]["rating"]
+
+    #if nothing in the watched list, return 0
+    else:
+        return 0
+
+    #calculate the average and return it
+    average = total/len(user_data["watched"])
+    return average
 
+def genres(user_data):
+    #set up set, dictionary for the count
+    genres = set()
+    
+    #make a set of genres, and return it as a list
+    for i in range(len(user_data["watched"])):
+        genres.add(user_data["watched"][i]["genre"])
+    return list(genres)
+
+def genre_count(user_data):
+    #set up set, dictionary for the count
+    user_genres = genres(user_data)
+    genre_count = {}
+
+    #iterate over the genres and add a count to a dictionary
+    for i in range(len(user_data["watched"])):
+        count = 0
+        for genre in user_genres:
+            if user_data["watched"][i]["genre"] == genre:
+                count += 1
+                genre_count[genre] = count
+        return genre_count
+
+def get_most_watched_genre(user_data):
+    #make sure the watched list has data
+    if user_data["watched"]:
+        watched_genre_count = genre_count(user_data)
+        most_watched = max(watched_genre_count, key=watched_genre_count.get)
+        return most_watched
+    
+    #if no data, return None
+    else:
+        return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_friend_data(user_data):
+    #make comprehensive list of all friend data
+    friend_data = []
+    for i in range(len(user_data["friends"])):
+        for j in range(len(user_data["friends"][i]["watched"])):
+            friend_data.append(user_data["friends"][i]["watched"][j])
+    return friend_data
+
+def get_unique_watched(user_data):
+    unique_watched = []
+    friend_data = get_friend_data(user_data)
+
+    #iterate over friend data and if movie isn't in friend data, add it to unique_watched
+    for i in range(len(user_data["watched"])):
+        if user_data["watched"][i] not in friend_data:
+            unique_watched.append(user_data["watched"][i])
 
-        
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    friends_unique_watched = []
+
+    #iterate over friends' watched lists, and if it's not in the user's list,
+    #append it to a new list and return it
+    for i in range(len(user_data["friends"])):
+        for j in range(len(user_data["friends"][i]["watched"])):
+            if user_data["friends"][i]["watched"][j] not in user_data["watched"]:
+                if user_data["friends"][i]["watched"][j] not in friends_unique_watched:
+                    friends_unique_watched.append(user_data["friends"][i]["watched"][j])
+    return friends_unique_watched

```

Comment:
We could use the helper function you wrote to help solve this function as well!
```suggestion
    friend_data = get_friend_data(user_data)
    friends_unique_watched = []

    #iterate over friends' watched lists, and if it's not in the user's list,
    #append it to a new list and return it
    for friend_movie in friend_data:
        if movie not in user_data['watched']:
            friends_unique_watched.append(movie)
    return friends_unique_watched

```